zlib-rs/src/c_api.rs

```diff
@@ -18,6 +18,39 @@ pub type voidp = *mut c_void;
 pub type voidpc = *const c_void;
 pub type voidpf = *mut c_void;
 
+/// The current stream state
+///
+/// # Custom allocators
+///
+/// The low-level API supports passing in a custom allocator as part of the [`z_stream`]:
+///
+/// ```no_check
+/// struct z_stream {
+///     // ...
+///     pub zalloc: Option<unsafe extern "C" fn(_: *mut c_void, _: c_uint, _: c_uint) -> *mut c_void>,
+///     pub zfree: Option<unsafe extern "C" fn(_: *mut c_void, _: *mut c_void)>,
```

Comment:
```suggestion
///     pub zalloc: Option<unsafe extern "C" fn(*mut c_void, c_uint, c_uint) -> *mut c_void>,
///     pub zfree: Option<unsafe extern "C" fn(*mut c_void, *mut c_void)>,
```

Rustc has been rejecting argument names in function pointer types since forever.